crates/nu-cli/src/types/deduction.rs

```diff
@@ -0,0 +1,1451 @@
+                        self.checked_insert(
+                            &VarUsage::new(var_name, &spanned_expr.span),
+                            VarShapeDeduction::from_usage_with_alternatives(
+                                &spanned_expr.span,
+                                &get_shapes_allowed_in_range(),
+                            ),
+                        )?;
+                    }
+                }
+                if let Some(range_right) = &range.right {
+                    if let Expression::Variable(Variable::Other(var_name, span)) = &range_right.expr
+                    {
+                        self.checked_insert(
+                            &VarUsage::new(&var_name, &spanned_expr.span),
+                            VarShapeDeduction::from_usage_with_alternatives(
+                                &span,
+                                &get_shapes_allowed_in_range(),
+                            ),
+                        )?;
+                    }
+                }
+            }
+            Expression::List(inner_exprs) => {
+                trace!("Infering vars in list");
+                for expr in inner_exprs {
+                    self.infer_shapes_in_expr((pipeline_idx, pipeline), expr, registry)?;
+                }
+            }
+            Expression::Invocation(invoc) => {
+                trace!("Infering vars in invocation: {:?}", invoc);
+                self.infer_shape(invoc, registry)?;
+            }
+            Expression::Table(header, _rows) => {
+                self.infer_shapes_in_table_header(header)?;
+                // Shapes within columns can be heterogenous as long as
+                // https://github.com/nushell/rfcs/pull/3
+                // didn't land
+                // self.infer_shapes_in_rows(_rows)?;
+            }
+            Expression::Variable(_) => {}
+            Expression::Literal(_) => {}
+            Expression::ExternalWord => {}
+            Expression::Synthetic(_) => {}
+            Expression::FilePath(_) => {}
+            Expression::ExternalCommand(_) => {}
+            Expression::Command => {}
+            Expression::Boolean(_) => {}
+            Expression::Garbage => {}
+        };
+
+        Ok(())
+    }
+
+    fn infer_shapes_in_table_header(
+        &mut self,
+        header: &[SpannedExpression],
+    ) -> Result<(), ShellError> {
+        for expr in header {
+            if let Expression::Variable(Variable::Other(name, _)) = &expr.expr {
+                let var = VarUsage::new(name, &expr.span);
+                self.checked_insert(
+                    &var,
+                    VarShapeDeduction::from_usage_with_alternatives(
+                        &var.span,
+                        &get_shapes_allowed_in_table_header(),
+                    ),
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    // Shapes within columns can be heterogenous as long as
```

Comment:
Was this section of commented code intended?

Reply:
It's deleted :) Thanks.